src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -279,36 +293,45 @@ def post_fork_child(self):
 
     # Broadcast our process group ID (in PID form - i.e. negated) to the remote client so
     # they can send signals (e.g. SIGINT) to all processes in the runners process group.
-    pid = str(os.getpgrp() * -1).encode('ascii')
-    NailgunProtocol.send_pid(self._socket, pid)
+    NailgunProtocol.send_pid(self._socket, os.getpgrp() * -1)
 
-    # Invoke a Pants run with stdio redirected and a proxied environment.
-    with self.nailgunned_stdio(self._socket, self._env) as finalizer,\
-         hermetic_environment_as(**self._env):
-      try:
-        # Setup the Exiter's finalizer.
-        self._exiter.set_finalizer(finalizer)
-
-        # Clean global state.
-        clean_global_runtime_state(reset_subsystem=True)
-
-        # Re-raise any deferred exceptions, if present.
-        self._raise_deferred_exc()
-
-        # Otherwise, conduct a normal run.
-        runner = LocalPantsRunner.create(
-          self._exiter,
-          self._args,
-          self._env,
-          self._target_roots,
-          self._graph_helper,
-          self._options_bootstrapper
-        )
-        runner.set_start_time(self._maybe_get_client_start_time_from_env(self._env))
-        runner.run()
-      except KeyboardInterrupt:
-        self._exiter.exit(1, msg='Interrupted by user.\n')
-      except Exception:
-        self._exiter.handle_unhandled_exception(add_newline=True)
-      else:
-        self._exiter.exit(0)
+    try:
+      # Invoke a Pants run with stdio redirected and a proxied environment.
+      with self.nailgunned_stdio(self._socket, self._env) as finalizer,\
+           hermetic_environment_as(**self._env):
+        try:
+          # Setup the Exiter's finalizer.
+          self._exiter.set_finalizer(finalizer)
+
+          # Clean global state.
+          clean_global_runtime_state(reset_subsystem=True)
+
+          # Re-raise any deferred exceptions, if present.
+          self._raise_deferred_exc()
+
+          # Otherwise, conduct a normal run.
+          runner = LocalPantsRunner.create(
+            self._exiter,
+            self._args,
+            self._env,
+            self._target_roots,
+            self._graph_helper,
+            self._options_bootstrapper
+          )
+          runner.set_start_time(self._maybe_get_client_start_time_from_env(self._env))
+          runner.run()
+        except KeyboardInterrupt:
+          self._exiter.exit(1, msg='Interrupted by user.\n')
+        except Exception:
+          self._exiter.handle_unhandled_exception(add_newline=True)
+        else:
+          self._exiter.exit(0)
+    except Exception:
```

Comment:
After logging it, this swallows the exception...it should probably re-raise it a bare `raise`. ie:
```
except Exception:
  do_stuff()
  raise
```

Reply:
Done in 02d0fa1 by adding a `re_raise` parameter to `handle_unhandled_exception()`! I thought this was a reasonable extension, let me know if that's ridiculous.

Reply:
It doesn't look like many callsites actually pass the `exc` parameter, and `re_raise` doesn't check that it has an `exc` to raise.

Reply:
You're right, I didn't realize the `exc_info()[1]` could return `None`. Will revise.